sycl/test/regression/operator-names.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -c %s -o %t.out -Wno-deprecated -fno-operator-names
+// RUN: %clangxx -fsycl -fsyntax-only %s -Wno-deprecated -fno-operator-names
```

Comment:
Line 7?

Reply:
> it does not appear to respect the -fno-operator-names option

What does it mean? Modified test passes.

I suggest we a use of operator name in the test itself and add --verify option with checks that operator names are used in the test, but no detected in the headers.

BTW, I suppose DPC++ extensions, which are not implicitly included by `<sycl/sycl.hpp>` should be tested as well. Right?

@Pennycook, FYI.

Reply:
It seems like the comment on line 7 is not correct. From my local experiments:

```
$ ./bin/clang++ -fsyntax-only t.cpp -fno-operator-names
t.cpp:2:13: error: expected ')'
  if (a > b and b >c ) {
            ^
t.cpp:2:6: note: to match this '('
  if (a > b and b >c ) {
     ^
1 error generated.
$ ./bin/clang++ -fsyntax-only t.cpp -foperator-names
$ echo $?
0
```

Reply:
Removed the comment in b29ae747f715bd47b67d1796613592402a6d7234